colors/build.js

```diff
@@ -15,6 +15,19 @@ function createColor(color, element, prefix) {
   return rv;
 }
 
+function createJsColor(color, element) {
+  const rv = [];
+
+  for (const variant in element) {
+    if (element.hasOwnProperty(variant)) {
+      const value = element[variant];
+      rv.push(`exports.${color.toUpperCase()}_${variant} = '${value}';\n`);
+    }
+  }
+  rv.push('\n');
+  return rv;
+}
+
```

Comment:
I wonder if it would be worth extracting out the looping logic, and passing in a `formatColor` callback instead having two very similar functions, one of which does some strange stuff sometimes…

Reply:
To me it seems more complicated to try and abstract them together. They both seem pretty simple as is.

Reply:
Yeah?  I thought it would be as simple as:
```diff
diff --git a/colors/build.js b/colors/build.js
index 5e8ff05..6b60f1a 100644
--- a/colors/build.js
+++ b/colors/build.js
@@ -2,13 +2,13 @@ const fs = require('fs');
 const colors = require('./photon.json');
 const metadata = require('./package.json');

-function createColor(color, element, prefix) {
+function createColor(color, element, formatter) {
   const rv = [];

   for (const variant in element) {
     if (element.hasOwnProperty(variant)) {
       const value = element[variant];
-      rv.push(`${prefix}${color}-${variant}: ${value};\n`);
+      rv.push(formatter(variant, value));
     }
   }
   rv.push('\n');
@@ -28,8 +28,12 @@ const sassOutput = [`/* Firefox Colors SCSS Variables v${metadata.version} * for (const color in colors) {
   if (colors.hasOwnProperty(color)) {
     const element = colors[color];
-    cssOutput.push(...createColor(color, element, '  --'));
-    sassOutput.push(...createColor(color, element, '$'));
+    cssOutput.push(...createColor(color, element,
+      (variant, value) => `  --${color}-${variant}: ${value};\n`);
+    sassOutput.push(...createColor(color, element,
+      (variant, value) => `$${color}-${variant}: ${value};\n`);
+    jsOutput.push(...createColor(color, element,
+      (variant, value) => `exports.${color.toUpperCase()}_${variant} = '${value}';\n`);
   }
 }```